### ISSUE TITLE
`prefer-toggle-attribute`: Should `required = true` be checked?

### ISSUE THREAD

**Issue by dangowans** [sindresorhus/eslint-plugin-unicorn#3751]
### Description

I'm wondering if `prefer-toggle-attribute` should help standardize the method of toggling attributes in a code base.

For example:

```javascript
inputElement.required = booleanCondition
```

Is essentially the same as:

```javascript
inputElement.toggleAttribute('required', booleanCondition)
```

Should the rule have an option to set the preferred "toggle style", or should the rule have a preferred style? Would a `consistent-toggle-attribute` rule make more sense?

### Examples

When `toggleStyle: 'function'`

```js
// ❌
inputElement.required = booleanCondition

// ✅
inputElement.toggleAttribute('required', booleanCondition)
```

When `toggleStyle: 'setter'`

```js
// ❌
inputElement.toggleAttribute('required', booleanCondition)

// ✅
inputElement.required = booleanCondition
```

**Comment by sindresorhus** [comment]
I don't think this is worth doing. `element.foo = condition` only matches `toggleAttribute('foo', condition)` for reflected boolean attributes on specific element types. For example, `input.checked` changes the state and not the attribute, `div.disabled` does not reflect, and SVG elements have no `hidden` property. The rule also cannot know if the receiver is an element, and `foo.required = x` is common on plain objects. So it would cause false positives and unsafe fixes.

I added a note to the docs about this.

**Comment by dangowans** [comment]
That's fair. Thanks for considering it, and the expert explanation @sindresorhus .
